### PR TITLE
Adjust public APIs to prefer 128-bit SST unique ID

### PR DESCRIPTION
**Pull request by pdillinger**
Summary: 128 bits should suffice almost always and for tracking in manifest.

Note that this changes the output of sst_dump --show_properties to only show 128 bits.

Also introduces InternalUniqueIdToHumanString for presenting internal IDs for debugging purposes.

Test Plan: unit tests updated

### CODE REVIEW

File: include/rocksdb/unique_id.h [hunk 1]
```diff
@@ -22,20 +20,33 @@ namespace ROCKSDB_NAMESPACE {
...
 // so that the "all zeros" value can be used reliably for a null ID.
-// Assuming one generates many SST files in the lifetime of each process,
-// the probability of collision between processes is "better than
-// random": if processes generate n SST files on average, we expect to
-// generate roughly 2^64 * sqrt(n) files before first collision in the
-// first 128 bits. See https://github.com/pdillinger/unique_id
-// Using the full 192 bits, we expect to generate roughly 2^96 * sqrt(n)
-// files before first collision.
+// These IDs are more than sufficient for SST uniqueness within a each of
+// meany DBs or hosts. For an extreme example assuming random IDs, consider
```
Comment (jay-zhuang) [inline include/rocksdb/unique_id.h:28]: typo?

File: include/rocksdb/unique_id.h [hunk 2]
```diff
@@ -22,20 +20,33 @@ namespace ROCKSDB_NAMESPACE {
...
+// appropriate to use this ID instead of the 128-bit ID if ID collisions
+// between files among any hosts in a vast fleet is a problem, such as a shared
+// global namespace for SST file backups. Under this criteria, the extreme
+// example above would expect a global file ID collision every 4 days with
+// 128-bit IDs (using some worst-case assumptions about process lifetime).
+// It's 10^17 years with 192-bit IDs.
+Status GetExtendedUniqueIdFromTableProperties(const TableProperties &props,
+                                              std::string *out_id);
+
 // EXPERIMENTAL: This API is subject to change
```
Comment (jay-zhuang) [inline include/rocksdb/unique_id.h:50]: should EXPERIMENTAL be removed?

**Review by jay-zhuang — APPROVED** [review]
LGTM

**Comment by pdillinger** [comment]
mini-crashtest in CircleCI saves the day again 😁

**Pull request closed without merging**